### ISSUE TITLE
NVIDIA Cloth: Selecting mesh node results in an error and an assert being printed in the Console

### ISSUE THREAD

**Issue by LB-KacperKapusta** [o3de/o3de#14695]
**Describe the bug**
Selecting Mesh node inside the Cloth component results in the following error being thrown along with the assert:
`[Error] (AzToolsFramework) - Out of range combo box index 1`

**File:**
`/home/gpc-589/o3de/Code/Framework/AzCore/./AzCore/std/containers/vector.h`

**Condition:**
```
(140641493209472) 'AZStd::vector::reference AZStd::vector<AZStd::pair<AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator>, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator>>>::operator[](AZStd::vector::size_type) [T = AZStd::pair<AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator>, AZStd::basic_string<char, AZStd::char_traits<char>, AZStd::allocator>>, Allocator = AZStd::allocator]'
(System) - AZStd::vector<>::at - position is out of range
```

**Line:**
`542`

The issue does reproduce for Chicken actor as well as other assets available as part of the Cloth component.
Please refer to the attached video and log for more information.

**Preconditions**

- Project with NVIDIA Cloth gem enabled is built.

**Steps to reproduce**
1. Create an entity with Actor and Cloth components.
2. Assign the chicken.actor asset in the Actor component.
3. Select the "chicken_mohawk" in the Mesh node field of the Cloth component.

**Expected behavior**
Mesh node is changed without any issues.

**Actual behavior**
Error and assert are printed when changing the mesh node of the Cloth component.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953659/220374033-5fa0a008-bf77-44af-b6a0-634cb13dd8ab.mp4

**Found in Branch**
Development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
[fde60c8](https://github.com/o3de/o3de/commit/fde60c8bb024abedee52abdcf07244b541c70b29)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version: 10
- CPU: AMD Ryzen 5 3600
- GPU: NVIDIA GeForce RTX 2060 SUPER
- Memory: 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/10794506/Editor.log)

**Comment by moraaar** [comment]
I could not reproduce this error with the latest code from today (commit 8b58313b37). Closing issue for now. 
@LB-KacperKapusta if this error happens again please reopen this bug and update the repro steps if necessary. Thanks!

**Comment by LB-KacperKapusta** [comment]
Issue verified as fixed in the Development (9c8d4f2).
